### PR TITLE
Redacted password in url while logging/metrics

### PR DESCRIPTION
**Pull request by sakshi-goyal-razorpay**
# Description
Redacted password in url while logging/metrics

## Type of change
- [x] Security fix 

# How Has This Been Tested?

- [x] Tested GetSubscription api on local/stage
- [x] Verified existing worker flow, and push endpoint logs
- [ ] Verified push endpoint label on grafana

# Is it a breaking change?
No

## Screenshots:
Screenshot 2022-08-29 at 1 41 33 PM
Screenshot 2022-08-29 at 1 41 52 PM

# Checklist:
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added unit tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added integration tests for the new feature (if applicable)
- [x] I have manually tested my code to the best of my abilities.

### CODE REVIEW

File: internal/subscription/model_test.go [hunk 1]
```diff
@@ -62,6 +62,7 @@ func Test_Model(t *testing.T) {
 	assert.False(t, dSubscription.IsPush())
 	assert.Nil(t, dSubscription.GetCredentials())
 	assert.False(t, dSubscription.HasCredentials())
+	assert.Empty(t, dSubscription.GetRedactedPushEndpoint())
```
Comment (subintp) [inline internal/subscription/model_test.go:65]: Can you add a test case for redacted pushendpoint ?
eg: redacted push endpoint != push endpoint if password is there 
       redacted push endpoint == push endpoint if there is no secret in the url

**Review by subintp — CHANGES_REQUESTED** [review]

File: internal/subscription/model_test.go (hunk 1 above)
Comment (sakshi-goyal-razorpay) [reply internal/subscription/model_test.go:65]: Sure

**Review by subintp — APPROVED** [review]

File: internal/subscription/model_test.go [hunk 2]
```diff
@@ -82,3 +80,24 @@ func Test_Model(t *testing.T) {
...
 	assert.Equal(t, "test-subscription.delay.5.seconds-subscriberID", dSubscription.GetDelayConsumerGroupInstanceID("subscriberID", "test-subscription.delay.5.seconds"))
 }
+
+func TestModel_GetRedactedPushEndpoint(t *testing.T) {
+	tests := []struct {
+		url    string
+		assert func(assert.TestingT, interface{}, interface{}, ...interface{}) bool
+	}{
+		{
+			url:    "https://google.com/",
```
Comment (vnktram) [inline internal/subscription/model_test.go:90]: Let's add more test cases where we have elaborate paths following the URI.

File: internal/subscription/model_test.go [hunk 3]
```diff
@@ -82,3 +80,24 @@ func Test_Model(t *testing.T) {
...
+			assert: assert.Equal,
+		},
+		{
+			url:    "https://username:password@google.com",
```
Comment (vnktram) [inline internal/subscription/model_test.go:94]: Can we also test malformed URLs that might still contain sensitive info?

**Review by vnktram — CHANGES_REQUESTED** [review]

**Review by vnktram — APPROVED** [review]

**Pull request merged**